src/components/LayerCollection/LayerPOLine.js

```diff
@@ -183,21 +182,27 @@ class LayerPOLine extends Component {
     });
   };
 
-  getCreatePOLIneInitialValues = (order) => {
+  getCreatePOLIneInitialValues = (order, vendor) => {
     const { parentResources } = this.props;
-    const { id: orderId, vendor: vendorId } = order;
-    const vendor = get(parentResources, 'vendors.records', []).find(d => d.id === vendorId);
+    const { id: orderId } = order;
     const createInventorySetting = getCreateInventorySetting(get(parentResources, ['createInventory', 'records'], []));
+    const initialCurrency = get(vendor, 'vendor_currencies[0]', 'USD');
+    const initialAccount = get(vendor, 'accounts[0].account_no', '');
+    const initialSubscriptionInterval = get(vendor, 'subscription_interval');
 
     const newObj = {
       source: {
         code: SOURCE_FOLIO_CODE,
       },
       cost: {
-        currency: CURRENCY.usd,
+        currency: initialCurrency,
       },
       vendorDetail: {
         instructions: '',
+        vendorAccount: initialAccount,
```

Comment:
IMHO, there is no much sense of different naming. `vendorAccount` should be good. The whole function is called like getInitialValues

Reply:
as for me it's not required to create variables above to pass them in object

Reply:
Fixed